hathor/transaction/base_transaction.py

```diff
@@ -424,8 +425,81 @@ def get_struct(self) -> bytes:
         struct_bytes += self.get_struct_nonce()
         return struct_bytes
 
+    def get_all_dependencies(self) -> Set[bytes]:
+        """Set of all tx-hashes needed to fully validate this tx, including parent blocks/txs and inputs."""
+        from hathor.transaction.genesis import GENESIS_HASHES
```

Comment:
Why do we remove the genesis? I feel we should return them just like any other dependency.

Reply:
Done.